server/src/main/java/org/elasticsearch/search/fetch/subphase/FetchFieldsPhase.java

```diff
@@ -71,6 +71,7 @@ public FetchSubPhaseProcessor getProcessor(FetchContext fetchContext) {
                         .stream()
                         .filter(
                             fieldAndFormat -> (searchExecutionContext.isMetadataField(fieldAndFormat.field) == false
+                                || searchExecutionContext.fieldExistsInIndex(fieldAndFormat.field) == false
                                 || searchExecutionContext.getFieldType(fieldAndFormat.field).isStored() == false
```

Comment:
`fieldExistsInIndex` checks if the field has a value in the current shard. This is not what we care about. We care if the field is mapped, which we check with `isFieldMapped`. Moreover, we care if the field is mapped only in the context of checking if the field is stored. That leads to this logic:

```suggestion
                                || (searchExecutionContext.isFieldMapped(fieldAndFormat.field) && searchExecutionContext.getFieldType(fieldAndFormat.field).isStored() == false)
```

Reply:
Got the idea. I had the impression that if the field is not available in the index, it should not be mapped and checking for `isStored` is not necessary anymore.